priv/posts/08-diagnostics-channel-is-landed.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Diagnostics Channel is landed!
+date: 2020-12-24 11:00
+tags: performance,nodejs,article,en-US
+---
+
+# Diagnostics Channel
+
+In `node` v15 was landed a feature that should help a lot of APM vendors. `diagnostics_channel` has the objective to have a centralized channel of events between modules.
+
+As blog release says:
+
+> _`diagnostics_channel` is a new experimental module that provides an API to create named channels to report arbitrary message data for diagnostics purposes.
+With `diagnostics_channel`, Node.js core and module authors can publish contextual data about what they are doing at a given time. This could be the hostname and query string of a mysql query, for example. Just create a named channel with dc.channel(name) and call channel.publish(data) to send the data to any listeners to that channel._
+
+This feature is similar to [`EventEmmiter`](https://nodejs.org/api/events.html#events_class_eventemitter), however is optimized enough to propagate data synchronous.
```

Comment:
`EventEmitter` is sync too. A more accurate statement would be that `diagnostics_channel` does have the inherent publishing cost that emitting a string-named event to an `EventEmitter` has. With `EventEmitter` there is a lookup cost to every single publish, mapping the string name of the event to the listener set. With `diagnostics_channel`, the separate object scoping means that the publish points _directly_ to the memory of the handler list.